### PR TITLE
Fix full no_std compatibility

### PR DESCRIPTION
**Pull request by CPerezz**
Resolves: #66

**NOTE** Will merge as soon as https://github.com/dusk-network/bls12_381/pull/55 is merged

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -19,19 +19,21 @@ exclude = [".github/workflows/ci.yml", "github/workflows/rust.yml",
 ]
 
 [dependencies]
-blake2 = "0.9"
 dusk-bytes = "0.1"
 dusk-bls12_381 = {version="0.6", default-features=false}
-subtle = {version="^2.3", default-features = false}
-rand_core = {version = "0.5", default-features=false}
+subtle = {version="2.0", default-features = false}
```
Comment (ZER0) [inline Cargo.toml:24]: Care to elaborate this downgrade…?
What are we gaining, what are we losing?

File: Cargo.toml [hunk 2]
```diff
@@ -19,19 +19,21 @@ exclude = [".github/workflows/ci.yml", "github/workflows/rust.yml",
...
+rand_core = {version = "0.6", default-features=false}
```
Comment (ZER0) [inline Cargo.toml:25]: does it needed to be updated? Why?

**Review by ZER0 — CHANGES_REQUESTED** [review]
Few questions to understand, plus I assuming this is solely needed at this moment due to dusk-plonk in no-std is that correct?

In that case, I will postpone the release of bls as well to the next iteration, however I would fix the broken_intra_doc_links (so I would create a different PR for that, as you did for jubjub)

File: Cargo.toml (hunk 2 above)
Comment (CPerezz) [reply Cargo.toml:25]: To make it no_std compatible with the new random crates. This seems to be the direction that Rust ecosystem is taking. Moving away from `rand` to `rand_core`.

File: Cargo.toml (hunk 1 above)
Comment (CPerezz) [reply Cargo.toml:24]: Nevermid. Will bring it back.

**Review by vlopes11 — APPROVED** [review]
LGTM

**Pull request merged**